src/runtime_src/core/tools/common/tests/TestIPU.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "TestIPU.h"
+#include "tools/common/XBUtilities.h"
+#include "xrt/xrt_bo.h"
+#include "xrt/xrt_device.h"
+#include "xrt/xrt_hw_context.h"
+#include "xrt/xrt_kernel.h"
+namespace XBU = XBUtilities;
+
+static constexpr size_t host_app = 1; //opcode
+static constexpr size_t buffer_size = 128;
+
+// ----- C L A S S   M E T H O D S -------------------------------------------
+TestIPU::TestIPU()
+  : TestRunner("ipu-verify", 
+                "Run 'Hello World' test on IPU",
+                "1x4.xclbin"){}
+
+boost::property_tree::ptree
+TestIPU::run(std::shared_ptr<xrt_core::device> dev)
+{
+  boost::property_tree::ptree ptree = get_test_header();
+
+  std::string xclbin_path;  
+  try {
+    xclbin_path = findXclbinPath(dev, ptree);
+  }
+  catch (const std::runtime_error& ex) {
+    logger(ptree, "Error", ex.what());
+    ptree.put("status", test_token_failed);
+    return ptree;
+  }
+
+  xrt::xclbin xclbin{xclbin_path};
+  if(!xclbin) { // if xclbin not found
+    return ptree;
+  }
```

Comment:
Line 37 throws if file not found.  Maybe include within earlier try?
```suggestion
```

Reply:
Fixed